src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.ValidateSignature.cs

```diff
@@ -143,19 +151,17 @@ private static SignatureValidationResult ValidateSignatureUsingAllKeys(
             bool kidMatched = configKidMatched || vpKidMatched;
 
             // No valid signature found. Return the exception details.
-            return new SignatureValidationResult(
-                ValidationFailureType.SignatureValidationFailed,
-                GetSignatureValidationFailureExceptionDetails(
-                    jwtToken,
-                    validationParameters,
-                    configuration,
-                    exceptionStrings,
-                    keysAttempted,
-                    kidExists,
-                    kidMatched));
+            return GetSignatureValidationError(
+                jwtToken,
+                validationParameters,
+                configuration,
+                exceptionStrings,
+                keysAttempted,
+                kidExists,
+                kidMatched);
         }
 
-        private static (SignatureValidationResult? validResult, bool KidMatched, KeyMatchFailedResult? failedResult) ValidateUsingKeys(
+        private static (Result<SecurityKey, ExceptionDetail>? validResult, bool KidMatched, KeyMatchFailedResult? failedResult) ValidateUsingKeys(
```

Comment:
Instead of the implicit value tuple, this would read a little better with an internal record struct, similar to what you have done elsewhere here.

Reply:
It would cause us to duplicate quite a bit of functionality between all of them as well.